dallinger/db.py

```diff
@@ -69,3 +70,33 @@ def init_db(drop_all=False):
     Base.metadata.create_all(bind=engine)
 
     return session
+
+
+def serialized(func):
+    """Run a function within a db transaction using SERIALIZABLE isolation.
```

Comment:
It looks like http://docs.sqlalchemy.org/en/latest/orm/session_state_management.html has session.dirty, session.deleted, etc for seeing what ORM operations are pending on a session. I'm not sure how that works with savepoints and the like, maybe it'd be best to explicitly roll back at the start so that it's consistent?